perceval/converters/abstract_converter.py

```diff
@@ -0,0 +1,147 @@
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+from abc import ABC, abstractmethod
+
+from perceval.components import Port, Circuit, Processor, Source
+from perceval.utils import P, BasicState, Encoding
+from perceval.utils.algorithms.optimize import optimize
+from perceval.utils.algorithms.norm import frobenius
+import perceval.components.unitary_components as comp
+
+MIN_PRECISION_GATE = 1e-4
+
+
+def _create_mode_map(c_idx: int, c_data: int) -> dict:
+    return {c_idx: 0, c_idx + 1: 1, c_data: 2, c_data + 1: 3}
+
+
+class AGateConverter(ABC):
+    r"""
+    Converter class for gate based Circuits to perceval processor
+    Qiskit or MyQLM
+    """
+
+    def __init__(self, catalog, **kwargs):
+        self._converted_processor = None
+        self._input_list = None  # input state in list
+        self._cnot_idx = 0  # counter for CNOTS in circuit
+        self._source = kwargs.get("source", Source())
+        self._backend_name = kwargs.get("backend_name", "SLOS")
+        self._heralded_cnot_builder = catalog["heralded cnot"]
+        self._heralded_cz_builder = catalog["heralded cz"]
+        self._postprocessed_cnot_builder = catalog["postprocessed cnot"]
+        self._generic_2mode_builder = catalog["generic 2 mode circuit"]
+        self._lower_phase_component = Circuit(2) // (0, comp.PS(P("phi2")))
+        self._upper_phase_component = Circuit(2) // (1, comp.PS(P("phi1")))
+        self._two_phase_component = Circuit(2) // (0, comp.PS(P("phi1"))) // (1, comp.PS(P("phi2")))
+
+    @property
+    @abstractmethod
+    def name(self) -> str:
+        """Each converter would have a distinct name as a string"""
+
+    @abstractmethod
+    def num_qbits_gate_circuit(self, gate_circuit) -> int:
+        pass
+
+    def configure_processor(self, gate_circuit, **kwargs):
+        """
+        Sets port Encoding and default input state for the Processor
+        """
+        qname = kwargs.get("qname", "Q")  # Default value, set any name provided by the gate circuit
+        n_qbits = self.num_qbits_gate_circuit(gate_circuit)
+        n_moi = n_qbits * 2  # number of modes of interest = 2 * number of qbits
+        self._input_list = [0] * n_moi
+        self._converted_processor = Processor(self._backend_name, n_moi, self._source)
+        for i in range(n_qbits):
+            self._converted_processor.add_port(i * 2, Port(Encoding.DUAL_RAIL, f'{qname}{i}'))
+            self._input_list[i * 2] = 1
+
+    def apply_input_state(self):
+        default_input_state = BasicState(self._input_list)
+        self._converted_processor.with_input(default_input_state)
+
+    @abstractmethod
+    def convert(self, gate_circuit, use_postselection: bool = True) -> Processor:
+        """
+        converts gates based circuits to one with linear optical components
+        and returns a perceval processor
+        Children - MyQlMConverter and QiskitConverter should define this method
```

Comment:
Please don't name children in a parent class. Being abstract means you need to be generic, and have new children without knowing about it.
Also, the info "Children should define this method" is already clear with the `@abstractmethod` modifier